coord/src/rmr/node.c

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Redis Ltd. 2016 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#include <string.h>
+
+#include "node.h"
+
+/* Return 1 both nodes have the same host */
+int MRNode_IsSameHost(MRClusterNode *n, MRClusterNode *other) {
+  if (!n || !other) return 0;
```

Comment:
Can this happened?

Reply:
I didn't touch this function, just moved it from node_map.c

`n` can't be NULL since we first check the node's flags and if it was NULL we would have crushed
```
if (strategy & MRCluster_MastersOnly && !(n->flags & MRNode_Master)) {
          continue;
        }
        if (!MRNode_IsSameHost(n, myNode)) {
          return n;
        }
```

and i guess `myNode` can't be NULL?